docs/addnewprovider.md

```diff
@@ -0,0 +1,56 @@
+# :memo: Adding support for a new provider
+
+### Step 1: Add provider specific options 
+
+Provider specific options goes under `cmd/cloud-api-adaptor`
+- [Options](https://github.com/confidential-containers/cloud-api-adaptor/blob/staging/cmd/cloud-api-adaptor/main.go#L48)
+- [Parsing](https://github.com/confidential-containers/cloud-api-adaptor/blob/staging/cmd/parse.go#L21)
+- [Calling the specific provider](https://github.com/confidential-containers/cloud-api-adaptor/blob/staging/cmd/cloud-api-adaptor/main.go#L103)
```

Comment:
Do we need to use permalinks here?

Reply:
> This documentation looks good explaining the current design.
> 
> I have a question about the current design. As described in Step2, separate binaries for each provider are generated using build tag. However, there are switch statements to check hypervisor type at runtime in main.go. I think we can eliminate these runtime checks using build tags.
> 
I was not aware of a way to eliminate these checks without having separate main files. If we can avoid that it'll be good.
So once the changes are there in the code, the same can be updated in this doc.

Reply:
> Do we need to use permalinks here?

Not really, but I thought its better to point to specific example. Do you have any other suggestion? Happy to change.

Reply:
I see. Let's merge this patch for now. We will update this doc when we change the code.